tests/modules/test_cloud.py

```diff
@@ -39,50 +39,103 @@ def assert_tensor_list_diff(a, b):
     assert all([all(_a != _b) for _a, _b in zip(a, b)])
 
 
+def mock_list_bucket_error_cmd(self, destination: str | None = None):
+    destination = (
+            Path(self.bucket) / destination
+            if destination is not None
+            else Path(self.bucket)
+        )
+    cmd = ["gsutil", "ls", f"gs://{destination}"]
+    raise Exception(f"There was an error running `{' '.join(cmd)}`. "
+            "Make sure gsutil is installed and that the destination exists. `CommandException: One or more URLs matched no objects.`")
+def mock_list_bucket_empty(self, destination: str | None = None):
+    return []
+
+
+def mock_rsync_up(
+        self,
+        local_path: Path,
+        remote_path: str,
+        logger: FileLogger | None = None,
+    ):
+    pass
+
 def test_gcp(tmp_path: Path, bucket: str = "gs://iordanis/"):
     rand_folder = f"{torch.rand(1).item()}"
     rand_destination = bucket + rand_folder
     # GcpConfig(bucket=rand_destination)._make_process(["gsutil", "-m", "rm", "-rf", rand_destination], verbose=False)
-    assert_error_msg(
-        lambda: GcpConfig(bucket=rand_destination),
-        f"There was an error running `gsutil ls {rand_destination}`. Make sure gsutil is installed and that the destination exists. `CommandException: One or more URLs matched no objects.`",
-    )
-    with mock.patch("socket.gethostname", return_value="localhost"):
+
+
+    with mock.patch("ablator.modules.storage.cloud.GcpConfig.list_bucket",mock_list_bucket_error_cmd):
         assert_error_msg(
-            lambda: GcpConfig(bucket=bucket),
-            f"Can only use GcpConfig from Google Cloud Server. Consider switching to RemoteConfig.",
+            lambda: GcpConfig(bucket=rand_destination),
+            f"There was an error running `gsutil ls {rand_destination}`. Make sure gsutil is installed and that the destination exists. `CommandException: One or more URLs matched no objects.`",
         )
-    cfg = GcpConfig(bucket=bucket)
-    files = cfg.list_bucket()
-    original_tensors = write_rand_tensors(tmp_path)
-    cfg.rsync_up(tmp_path, rand_folder)
-
-    new_files = cfg.list_bucket()
+    with mock.patch("ablator.modules.storage.cloud.GcpConfig._find_gcp_nodes",return_value={}):
```

Comment:
Can you please mock non-empty gcp_nodes? i.e. use different directories

Reply:
The host name is localhost here so I think the nodes are expected to be null.